code/game/objects/structures/industrial_lift.dm

```diff
@@ -227,16 +230,36 @@ GLOBAL_LIST_EMPTY(lifts)
 		destination = get_step_multiz(src, going)
 	else
 		destination = going
+	if(istype(destination, /turf/closed/wall))
+		var/turf/closed/wall/C = destination
+		do_sparks(2, FALSE, C)
+		C.dismantle_wall(devastated = TRUE)
+		for(var/mob/M in urange(8, src))
+			shake_camera(M, 2, 3)
+		
+		playsound(C, 'sound/effects/meteorimpact.ogg', 100, TRUE)
 	if(going == DOWN)
 		for(var/mob/living/crushed in destination.contents)
 			to_chat(crushed, "<span class='userdanger'>You are crushed by [src]!</span>")
 			crushed.gib(FALSE,FALSE,FALSE)//the nicest kind of gibbing, keeping everything intact.
 	else if(going != UP) //can't really crush something upwards
-		for(var/obj/structure/anchortrouble in destination.contents)
-			if(!QDELETED(anchortrouble) && anchortrouble.anchored && (!istype(anchortrouble, /obj/structure/holosign)) && anchortrouble.layer >= GAS_PUMP_LAYER) //to avoid pipes, wires, etc
+		var/atom/throw_target = get_edge_target_turf(src, turn(going, pick(45, -45))) //finds a spot to throw the victim at for daring to be hit by a tram
+		for(var/obj/structure/victimstructure in destination.contents)
+			if(!QDELETED(victimstructure) && (!istype(victimstructure, /obj/structure/holosign)) && victimstructure.layer >= GAS_PUMP_LAYER) //to avoid wires
+				playsound(src, 'sound/effects/bang.ogg', 50, TRUE)
+				if(victimstructure.anchored && initial(victimstructure.anchored) == TRUE) //prevents you from wrenching lockers/crates to easily break them open
+					visible_message("<span class='danger'>[src] smashes through [victimstructure]!</span>")
+					victimstructure.deconstruct(FALSE)
+				else
+					visible_message("<span class='danger'>[src] violently rams [victimstructure] out of the way!</span>") 
+					victimstructure.anchored = FALSE
+					victimstructure.take_damage(rand(20,25))
+					victimstructure.throw_at(throw_target, 200, 4)
+		for(var/obj/machinery/victimmachine in destination.contents)
+			if(!QDELETED(victimmachine) && victimmachine.layer >= GAS_PUMP_LAYER) //to avoid pipes
 				playsound(src, 'sound/effects/bang.ogg', 50, TRUE)
-				visible_message("<span class='notice'>[src] smashes through [anchortrouble]!</span>")
-				anchortrouble.deconstruct(FALSE)
+				visible_message("<span class='danger'>[src] smashes through [victimmachine]!</span>")
+				victimmachine.Destroy()
```

Comment:
Why Destroy()? You mean qdel() or deconstruct()?

Reply:
Destroy doesn't leave a frame like deconstruct does, and it seems like deconstruct checks to see if it doesn't have the NODECONSTRUCT_1 flag.

Reply:
Destroy() is only called by qdel() itself no exceptions. just use qdel

Reply:
use qdel